app/src/main/java/com/zeapo/pwdstore/UserPreference.kt

```diff
@@ -597,12 +628,18 @@ class UserPreference : AppCompatActivity() {
                         setResult(Activity.RESULT_OK)
 
                         finish()
-                    } catch (e: IOException) {
-                        MaterialAlertDialogBuilder(this)
-                                .setTitle(this.resources.getString(R.string.ssh_key_error_dialog_title))
-                                .setMessage(this.resources.getString(R.string.ssh_key_error_dialog_text) + e.message)
-                                .setPositiveButton(this.resources.getString(R.string.dialog_ok), null)
-                                .show()
+                    } catch (e: Exception) {
+                        when (e) {
+                            is IOException,
+                            is IllegalArgumentException -> {
+                                MaterialAlertDialogBuilder(this)
+                                        .setTitle(this.resources.getString(R.string.ssh_key_error_dialog_title))
+                                        .setMessage(this.resources.getString(R.string.ssh_key_error_dialog_text) + e.message)
+                                        .setPositiveButton(this.resources.getString(R.string.dialog_ok), null)
+                                        .show()
+                            }
+                            else -> throw e
```

Comment:
This can't be happening. Feel free to differentiate the message based on the exception type but we don't wanna ever crash the process.